### DISCUSSION TITLE
How to debug "ERROR in SERVER PRERENDERING" when there is no output

### DISCUSSION THREAD

**Discussion by JaredShay** [shakacode/react_on_rails#1550] [Q&A] [answered]
I've recently added `react_on_rails` to an application and am slowly rolling out SSR components. There have been two errors with the following:

```
ActionView::Template::Error: ERROR in SERVER PRERENDERING
Encountered error:

Error evaluating server bundle. Check your webpack configuration.
===============================================================
Caught error:

================================

when prerendering ProMarketingCTAs with props: {"shouldOverlapPaywall":true}

code:

        (function() {
          var railsContext = {"railsEnv":"production","inMailer":false,"i18nLocale":"en","i18nDefaultLocale":"en","rorVersion":"13.3.4","rorPro":false,"href":"https://www.theinformation.com/org-charts/snap","location":"/org-charts/snap","scheme":"https","host":"www.theinformation.com","port":null,"pathname":"/org-charts/snap","search":null,"httpAcceptLanguage":null,"serverSide":true};
              ReactOnRails.clearHydratedStores();

          var props = {"shouldOverlapPaywall":true};
          return ReactOnRails.serverRenderReactComponent({
            name: 'ProMarketingCTAs',
            domNodeId: 'ProMarketingCTAs-react-component-9524e97b-5d8d-4ee6-9e10-4c744f528be9',
            props: props,
            trace: false,
            railsContext: railsContext
          });
        })()

```

The `Caught error:` is showing the output from a call within the `execjs` gem:

```ruby
def exec_runtime(filename)
  io = IO.popen(binary.split(' ') << filename, **(@popen_options.merge({err: [:child, :out]})))
  output = io.read
  io.close

  if $?.success?
    output
  else
    raise exec_runtime_error(output) # this is the line that is raising, output is `nil` or `''`
  end
end
```

And the component and function referenced appear to work correctly. The page where this error has occurred appears to work. It is from a reasonably high traffic site and this error report appears to be very intermittent and rare. 

I can't reproduce this so I can't file a bug report. Does anyone have suggestions for what to do to debug this further? My main concern is if I scale up the usage of SSR I'm not sure how to diagnose issues when I'm getting no output.

**Comment by justin808** [comment]
@JaredShay you have an issue in your webpack configuration to build a server bundle.

You might post to our community [Slack](https://join.slack.com/t/reactrails/shared_invite/enQtNjY3NTczMjczNzYxLTlmYjdiZmY3MTVlMzU2YWE0OWM0MzNiZDI0MzdkZGFiZTFkYTFkOGVjODBmOWEyYWQ3MzA2NGE1YWJjNmVlMGE)

If you'd like ShakaCode to help you out so that you can focus on new features as opposed to configuring Webpack, please [book a time with me](https://meetings.hubspot.com/justingordon/30-minute-consultation) or email me: justin@shakacode.com.

**Comment by Judahmeek** [comment, marked as answer]
@JaredShay try ensuring that you're using `NodeJS` for your `EXECJS_RUNTIME`.

`MiniRacer` isn't compatible with certain libraries & javascript features.

Additionally, there's a possibility that this is being caused by a new webserver worker getting a SSR request before the server-bundle has been compiled.

At least, I think that's an edge case we've encountered before. I'm having trouble tracking down any details on it, however.
